src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -199,9 +199,14 @@ public MixpanelPushNotification(Context context, Notification.Builder builder, l
         mData.setSticky(sticky);
 
         int notificationIcon = MixpanelNotificationData.NOT_SET;
+
+        ResourceIds resource = mDrawableIds;
         if (iconName != null) {
             if (mDrawableIds.knownIdName(iconName)) {
-                notificationIcon = mDrawableIds.idFromName(iconName);
+                notificationIcon = resource.idFromName(iconName);
+            } else if ((resource = getResourceIds(ResourceReader.MIPMAP_TYPE, mContext)).knownIdName(iconName)) {
```

Comment:
so do you think it'd make sense to do this for all ResourceReader.*? I'm genuinely asking because I don't know what developers would expect here.

Reply:
No, I don't think so. Other types include layout, color, string, navigation, animation, and animator so when creating an icon resource I think it makes sense for it to be covered by these two types. I think a large majority will be in drawable and experience no issues and the rest probably still use mipmap from older iterations of their code.

Reply:
It makes sense to do it in ResourceReader. My proposal is to have something along these lines:

```
this.mDrawableIds = getResourceIds(ResourceReader.DRAWABLE_TYPE, context);
this.mMipmapIds = getResourceIds(ResourceReader.MIPMAP_TYPE, context);

...

if (mDrawableIds.knownIdName(iconName)) {
  //
} else if (mMipmapIds.knownIdName(iconName)) {
   //
}
```

Ideally, `ResourceReader` should be refactored so [getLocalClassName](https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/mpmetrics/ResourceReader.java#L113) can return a list of classes (`getLocalClassNames`) and we don't have to check for both drawables and mipmaps everytime we want to find an image asset.

For example, you forgot to check for a mipmap image [in this function](https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java#L287) and the above solution would have covered that.